lib/serializer.ts

```diff
@@ -578,7 +584,7 @@ function deserializeCompositeType(serializer: Serializer, mapper: CompositeMappe
       if (Array.isArray(responseBody[key]) && modelProps[key].serializedName === "") {
         propertyInstance = responseBody[key];
         instance = serializer.deserialize(propertyMapper, propertyInstance, propertyObjectName);
-      } else if (propertyInstance !== null && propertyInstance !== undefined) {
+      } else if (propertyInstance !== undefined) {
```

Comment:
`propertyInstance != undefined`?

Reply:
I think I had a particular case here where I wanted to include a null value on the `instance` when deserializing, but drop anything that came back `undefined`.

One place where this would happen is if the property didn't exist on the parsed object tree, but we descended into `responseBody[key]` so we could check if it meets the required/nullable criteria for that property's spec.